source/utils/cache.js

```diff
@@ -24,6 +23,21 @@ cache.resolveFunc = (key) => {
 }
 
 /**
+ * @function getHardValue
+ * @description In Linux, or certain settings, it seems that cached promises
+ *   are not able to resolved and we need to cache raw result of promieses.
+ * @param {prom} - raw value or promise to be returned or resolved
+ * @param {promise} - a promise where next "then" will result in raw value.
+ */
+const getHardValue = (prom) => {
+  if (prom.then) {
+    return prom.then(getHardValue);
+  } else {
+    return Bluebird.resolve(prom);
```

Comment:
Strange... normally [`.resolve`](http://bluebirdjs.com/docs/api/promise.resolve.html) should already do this:
> If value is a thenable (Promise-like object, like those returned by jQuery's $.ajax), returns a trusted Promise that assimilates the state of the thenable.